sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -0,0 +1,277 @@
+#include "sw/device/silicon_creator/lib/shutdown.h"
+
+#include <assert.h>
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/alert.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+
+#include "alert_handler_regs.h"
+#include "flash_ctrl_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "keymgr_regs.h"
+#include "otp_ctrl_regs.h"
+#include "sram_ctrl_regs.h"
+
+
+static_assert(
+  ALERT_HANDLER_ALERT_CLASS_MULTIREG_COUNT <= OTP_CTRL_PARAM_ROM_ALERT_CLASSIFICATION_SIZE / 4,
+  "More alerts than alert classification OTP words!");
+
+
+#ifdef MOCK_ABS_MMIO
+// If we're building as a unit test, rename the shutdown functions so they
+// can be mocked and/or tested individually.
+#define SHUTDOWN_FUNC(name_) \
+  extern void name_;         \
+  void internal_##name_
+#else
+#define SHUTDOWN_FUNC(name_) static inline void name_
```

Comment:
Nit: `static inline` doesn't actually do anything different from `static`. (Also, I can't quite remember but I think we banned `static inline` in the style guide because it's a wierd GCCism?)

Reply:
Hm, I don't mean whatever GCCism `static inline` would refer to.  I mean static linkage and I require inlining.

I've added an ALWAYS_INLINE directive to `macros.h` which includes `__attribte__((always_inline))`.   Is that satisfactory?

Reply:
> I've added an ALWAYS_INLINE directive to macros.h which includes `__attribute__((always_inline))`. Is that satisfactory?

SGTM

Reply:
I don't particularly love `always_inline`, but fine, I guess.

`static inline` doesn't mean what you think it means anyway. `static` implies `inline`.